### PR TITLE
Add a condition to display Subscription tab

### PR DESCRIPTION
**Pull request by Quetzacoalt91**

### CODE REVIEW

**Review by ClaraLpresta — APPROVED** [review]

File: _dev/apps/ui/src/App.vue [hunk 1]
```diff
@@ -38,6 +38,7 @@
           </MenuItem>
           <MenuItem
             :route="{name: 'Billing'}"
+            v-if="GET_BILLING_SUBSCRIPTION_ACTIVE"
```
Comment (kktos) [inline _dev/apps/ui/src/App.vue:41]: Is this variable a flag ? a boolean ?
if it is, it should be better to have a name that reflects it, as isGetBillingSubActive or hasGetBillingSub....

**Review by kktos — CHANGES_REQUESTED** [review]

File: _dev/apps/ui/src/App.vue (hunk 1 above)
Comment (Quetzacoalt91) [reply _dev/apps/ui/src/App.vue:41]: It's a boolean, and thanks to the `mapGetter` the name of the getter we can directly use without proxy variable.

Comment (kktos) [reply _dev/apps/ui/src/App.vue:41]: but where is it defined ?

Comment (Quetzacoalt91) [reply _dev/apps/ui/src/App.vue:41]: This is likely because I'm not aiming for `master` here. Were you on the branch `dev/monetization`?

Comment (kktos) [reply _dev/apps/ui/src/App.vue:41]: oh. ok, different branch.

Comment (Quetzacoalt91) [reply _dev/apps/ui/src/App.vue:41]: No it's still scoped in the component. `mapGetters` is a helper able to use the store available in the component to return the getters provided in an array.

Comment (kktos) [reply _dev/apps/ui/src/App.vue:41]: ok, now I got it.
so, no, it's not a boolean, is an object's method.
and I reckon this code is only testing if the method is defined or not.
Now, it raises the question why do we need to test whether it's defined.....
I reckon it should be always defined, no ?
Or is this available behind a feature flag ?

Comment (Quetzacoalt91) [reply _dev/apps/ui/src/App.vue:41]: `mapGetters` will make the name of the getter always available to the component make it return the result of the associated method in https://github.com/PrestaShopCorp/psxmarketingwithgoogle/blob/ba7c19b415adc5480ab5b7b71080376f8e8296c9/_dev/apps/ui/src/store/modules/app/getters.ts#L121-L125

The type of `GET_BILLING_SUBSCRIPTION_ACTIVE` is always a boolean is this line.

Comment (kktos) [reply _dev/apps/ui/src/App.vue:41]: ok. so back to my initial comment :
this getter name, as it is a boolean, should have a name that shows it; like
IS_BILLING_SUBSCRIPTION_ACTIVE
or
HAS_BILLING_SUBSCRIPTION

Comment (tberger-prestashop) [reply _dev/apps/ui/src/App.vue:41]: De ce que je comprends, GET_BILLING_SUBSCRIPTION_ACTIVE est bien une méthode "(LocalState): boolean" qui est appelée au runtime

Comment (kktos) [reply _dev/apps/ui/src/App.vue:41]: nope. c'est un getter vuex.
C'est un peu le même principe pour une prop en js. Exemple:
```javascript
const MyObject = {
    firstname: "Terry",
    lastname: "Berger",
    get name() {
        return this.firstname + "_"+ this.lastname
    }
};

console.log( myObject.name );
// Terry_Berger
```

**Pull request merged**